src/Collection.js

```diff
@@ -298,26 +298,80 @@ var CPt = Collection.prototype;
  */
 function registerMethods(methods, type) {
   for (var methodName in methods) {
-    if (CPt.hasOwnProperty(methodName)) {
-      throw Error(`A method with name "${methodName}" already exists.`);
+    if (!methods.hasOwnProperty(methodName)) {
+      return;
+    }
+    if (hasConflictingRegistration(methodName, type)) {
+      throw Error(`There is a conflicting registration for method with name "${methodName}".`);
```

Comment:
What do you think, would it be valuable to provide more information and mention the types which already have a registered method with that name?